### DISCUSSION TITLE
Bypass age restriction

### DISCUSSION THREAD

**Discussion by FabrisNick** [ReVanced/revanced-discussions#118] [Question]
It would be great if ReVanced could bypass the age restrictions.

**Comment by oSumAtrIX** [comment]
Definitely a good idea, it should be possible, theres that one extension which can do that on the browser.

**Comment by FabrisNick** [comment]
> Definitely a good idea, it should be possible, theres that one extension which can do that on the browser.

Yeah, I use it with Tampermonkey 

https://github.com/zerodytrash/Simple-YouTube-Age-Restriction-Bypass/

**Reply by cutekita** [reply]
still works today

**Reply by PukemStack** [reply]
doesn't work today

**Comment by pukkandan** [comment]
Not sure how easy/difficult this is to implement in re-vanced, but here's the details of what that extension does:

Youtube videos can be roughly categorized into 4 tiers based on their embeddability. See https://github.com/yt-dlp/yt-dlp/pull/575#issuecomment-888837000. Age-gated videos in tiers 1-3 can be bypassed by spoofing certain "clients" in the INNERTUBE requests
* tier 1 can be bypassed with `ANDROID_EMBEDDED_PLAYER`/`WEB_EMBEDDED_PLAYER`
* We used to be able to bypass tiers 2-3 by setting `clientScreen=EMBED` in the request (Ref: https://github.com/yt-dlp/yt-dlp/pull/575) but this bypass **is now patched**
* Currently, tiers 1-3 can by bypassed by using the `TVHTML5_SIMPLY_EMBEDDED_PLAYER` client (Ref: https://github.com/yt-dlp/yt-dlp/pull/3233)
* None of these solve the account verification requirement inside EU. But that requirement can be bypassed by using `WEB_CREATOR` (`ANDROID_CREATOR` also works, but has less formats than other clients) **IF the user is signed in with an unverified youtube account**. Afaik, there is no client that can bypass this without an account

There is no known client-side bypass for tier 4 videos (without logging in). The only way to circumvent them is to [use an account proxy](https://github.com/zerodytrash/Simple-YouTube-Age-Restriction-Bypass/tree/main/account-proxy). Note that some videos like `gKVjWNv9V8Q` has a `gcr` parameter in the playback URL. These can only be accessed from the same country (by IP) as where the link was generated. So, for such videos, the playback will need to be performed [over a proxy](https://github.com/zerodytrash/Simple-YouTube-Age-Restriction-Bypass/blob/7a48b2e426e0e4123688ca4f3f0c5cb4ff1eeccb/dist/Simple-YouTube-Age-Restriction-Bypass.user.js#L50) in the same country as the account proxy. Since this solution needs third party services to work, it is understandable if the devs are unwilling to accept this.

Parameters for the above-mentioned clients: https://github.com/yt-dlp/yt-dlp/blob/e6f868a63c15f576152733a1508f474b5e5bd1ef/yt_dlp/extractor/youtube.py#L77-L233
More info about various clients: https://github.com/zerodytrash/YouTube-Internal-Clients

Correction: `TVHTML5_SIMPLY_EMBEDDED_PLAYER` works to bypass tiers 1-3 even in EU. `WEB_CREATOR` is only needed to access tier 4 videos

**Comment by Sculas** [comment]
What is the use case for this feature? **Unauthenticated** watching of age gated video's?

**Reply by RobinRMC** [reply]
> What is the use case for this feature? **Unauthenticated** watching of age gated video's?

YouTube also age-restrict videos that aren't even inappropriate. Look at the channel of [lukeafk](https://www.youtube.com/channel/UCw1KJiZ0M85ehyEKcbjN_nQ/videos) for example. A few of his videos are age-restricted, while there is no inappropriate content.

**Reply by Sculas** [reply]
I already got a reply for this in the Discord a while ago.

Lately, YouTube requires users to send them their IDs and/or credit cards before they're allowed to watch age-restricted videos. Understandably, a lot of users don't want to do that. That's the motivation behind this suggestion.

To my knowledge, the API blocks the request for age-restricted videos if a user is either unauthenticated or does not have permission to watch age-restricted videos, so bypassing this will be quite hard to do, especially by means of patching. "We'd have to get the big guns out", so to say.

To respond to @RobinRMC's message, does the above also apply to you? **Or if not**, why would you want this if you're allowed to watch age-restricted videos anyway?

**Reply by pukkandan** [reply]
> Lately, YouTube requires users to send them their IDs and/or credit cards before they're allowed to watch age-restricted videos.

Note that this happens only in the EU (ie, if your account is EU based - VPN don't help)

**Comment by FabrisNick** [comment]
> What is the use case for this feature? **Unauthenticated** watching of age gated video's?

Also because i don't want to provide my credit card or my identity card to watch age restricted videos.

**Comment by Degl1** [comment]
Would be nice if you could use this method to bypass the children's videos. Because children's videos don't play in the miniplayer.

**Reply by emilypeto** [reply]
Especially a pain because sometimes random videos are misidentified as being for children

**Reply by Sculas** [reply]
The YT kids shit is a pain in the ass anyway, we'll get rid of it entirely. 9 year olds don't know how to install ReVanced anyway, so it'll be fine :')

**Reply by fillwithjoy1** [reply]
Isn't the miniplayer pausing a client side fix? Should be able to hook into the video player and cut off the code responsible for auto pausing

**Reply by oSumAtrIX** [reply]
That's already present as a patch

**Reply by cutekita** [reply]
unfortunately it isn't. there's only bypassing the viewer discretion popup. I still have to copy the link and open it in a browser running the user script to watch videos with ID collection

**Reply by oSumAtrIX** [reply]
It is and the video is not being paused

**Reply by cutekita** [reply]
ill check again

edit 1; unless you mean kids bypass, which i think would be weird to select as the answer to the main question
edit 2; it still blocks age restricted videos
